src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,239 @@
+ * calculate) then interpolate a constant period trajectory. This is valuable for discrete
+ * time motion controllers that operate only at constant delta-time intervals
+ *
+ * @pre variable_time_trajectory is a valid (obeys the physical limitation of the robot)
+ * trajectory
+ * @pre constant_period_trajectory is pre=allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param constant_period_trajectory [out] The array to be filled with the constant time
+ * interpolation period equivalent of the input trajectory
+ * @param variable_time_trajectory [in] The valid trajectory used as
+ * reference for a constant interpolation period trajectory
+ * @param interpolation_period [in] The constant change in time that corresponds to each
+ * trajectory segment
+ *
+ */
+void app_trajectory_planner_interpolate_constant_time_trajectory_segmentation(
+    Trajectory_t* constant_period_trajectory, Trajectory_t* variable_time_trajectory,
+    const float interpolation_period);
+
+
+/**
+ * Returns a the total length of any arc specified by an arc parameterizaton
+ *
+ * @param arc_length_param [in] Arc length parameterization
+ *
+ * @return [out] The arc length in meters
+ */
+static float app_trajectory_planner_get_total_arcLength(
+    ArcLengthParametrization_t arc_length_param);
+
+/**
+ * Returns the maximum velocity profile for a given curve based on curvature and maximum
+ * allowable acceleration
+ *
+ * @pre max_allowable_speed_profile is pre-allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param max_allowable_speed_profile [out] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on th curve
+ * @param traj_elements [in] Trajectory segments including X/Y position
+ *
+ * @param path [in] The polynomial that defines the path of the trajectory
+ *
+ * @param num_elements [in] The number of elements in the trajectory
+ *
+ * @param arc_length_param [in] The arc length parameterization of the path polynomial
+ *
+ * @param arc_segment_length [in] The length of each segment in the trajectory
+ *
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any discrete
+ * point on the trajectory
+ *
+ */
+static void app_trajectory_planner_get_max_allowable_speed_profile(
+    float* max_allowable_speed_profile, TrajectoryElement_t* traj_elements,
+    Polynomial2dOrder3_t path, unsigned int num_elements,
+    ArcLengthParametrization_t arc_length_param, float arc_segment_length,
+    const float max_allowable_acceleration);
+
+/**
+ * Builds the forwards continuous velocity profile for the given parameters.
+ * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
+ * not guarantee a feasible path to follow.
+ *
+ * NOTE: The velocity_profile wil contain the same number of real elements as the input
```

Comment:
What is a "real element"?

Reply:
Not sure why I put 'real' it will just have the same number of elements